src/lib/block_producer/block_producer.ml

```diff
@@ -1295,43 +1210,52 @@ let run ~context:(module Context : CONTEXT) ~vrf_evaluator ~prover ~verifier
                 "Block producer will begin producing only empty blocks after \
                  $slot_diff slots"
               slot_tx_end ;
-            let next_check_now = check_next_block_timing new_global_slot i' in
             (* TODO: Re-enable this assertion when it doesn't fail dev demos
              *       (see #5354)
              * assert (
                    Consensus.Hooks.required_local_state_sync
                     ~constants:consensus_constants ~consensus_state
                     ~local_state:consensus_local_state
                    = None ) ; *)
-            let dispatch_now triple =
-              ignore
-                ( Interruptible.finally
-                    (Singleton_supervisor.dispatch production_supervisor triple)
-                    ~f:next_check_now
-                  : (_, _) Interruptible.t )
+            let next_check_now () = return (new_global_slot, i', prev_step) in
+            let dispatch_now data =
+              Option.iter !prev_step ~f:(fun ivar ->
+                  if Ivar.is_full ivar then
+                    [%log error] "Ivar.fill bug is here!" ;
+                  Ivar.fill ivar () ) ;
+              let intr_ivar = Ivar.create () in
+              let this_step = ref (Some intr_ivar) in
+              let produce_intr =
+                let%map.Interruptible x = produce intr_ivar data in
+                this_step := None ;
+                x
+              in
+              let%map _ = Interruptible.force produce_intr in
+              (* TODO consider uncommenting below or removing interruptible usage completely *)
```

Comment:
This comment is inaccurate (and suggested code has no effect)

Forcing `produce_intr` makes sure that the next iteration won't start before the block is produced. Hence interruptible will not ever be actually interrupted.

Reply:
Is there anything you want to do about this in this MR ?

Reply:
Probably not. Maybe it;'s worth copying as a code comment. Here I mostly explained what happens to myself 🗡️ 

CC @anne-laure-s

Reply:
Ah, this TODO is removed in PR chain, so no need to port it to code